scipy/sparse/linalg/isolve/tests/test_minres.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from numpy.testing import (assert_equal, assert_array_equal, assert_allclose,
+                           assert_raises, run_module_suite, dec)
+from scipy.sparse import diags, eye, lil_matrix
+from scipy.sparse.linalg.isolve import minres
+
+from .test_iterative import assert_normclose
+
+
+def get_sample_problem():
+    # A Random 10 x 10 sparse matrix
+    np.random.seed(1234)
+    matrix = diags(np.random.rand(10), format='csr')
+    # A Random Vector of length 10
+    np.random.seed(1234)
+    vector = np.random.rand(10)
+    return (matrix, vector)
+
+
+def test_singular():
+    A, b = get_sample_problem()
+    A[(0, 0)], b[0] = 0, 0
```

Comment:
For the full matrix you will need to do more than setting a diagonal element to 0, like setting a full row to zeros.

Also you can write simply `A[0, 0]`.

Reply:
Also can we test a situation when the system is inconsistent (without setting b[0] to 0)? In this case some sort of least-squares solution with a small norm should be returned. I'm not sure how to test it.

Reply:
I found a way to test this, I'll update the commit soon.

Reply:
This isn't nice. Does it mean that the function is only half-working? 

I tried `b[0] = 0.1` and `maxiter=1000`. Then it returns `info=0`, but `A xp != b` in all components. Can you create an issue for that?

Overall the quality of this function is dubious.

Reply:
Yeah, that's weird.
 I'll get back on this tomorrow.
Feel free to create an issue. (If you want)